### DISCUSSION TITLE
adding support for the chatglm3-6b model

### DISCUSSION THREAD

**Discussion by mnlife** [ggml-org/llama.cpp#6623] [Q&A]
Hello everyone, I'm currently adding support for the chatglm3-6b model on llama.cpp. I've done a quick check on the inference algorithm, but I didn't find any issues. 
However, the inference results are all over the place. 

How should I go about debugging this?
Are there any tricks for pinpointing the issue when inference goes wrong?

The following are my log link and code modifications:

[main.log](https://github.com/ggerganov/llama.cpp/files/14953319/main.log)
[0001-feature-add-chatglm-support.patch.txt](https://github.com/ggerganov/llama.cpp/files/14953334/0001-feature-add-chatglm-support.patch.txt)

**Comment by ggerganov** [comment]
Debugging could be tedious - you can try to compare the results after each op with the reference implementation. Or open a PR and more people can have a look

From a quick look, the following code incorrectly passes `split_point` for the `offset` argument of `ggml_view_2d` which expects bytes instead of elements:

```c++
+                // 创建两个视图张量，分别表示切分后的两部分
+                int64_t split_point = cur->ne[0] / 2;
+                struct ggml_tensor * x0 = ggml_cont(ctx, ggml_view_2d(ctx, cur, split_point, cur->ne[1], cur->nb[1], 0));
+                struct ggml_tensor * x1 = ggml_cont(ctx, ggml_view_2d(ctx, cur, split_point, cur->ne[1], cur->nb[1], split_point));
```

**Reply by mnlife** [reply]
Thank you!!!
I've made modifications to this, along with some other changes. After testing, everything seems fine. I'll tidy it up and push this commit shortly.
